yarn-project/pxe/src/kernel_prover/kernel_prover.ts

```diff
@@ -118,25 +117,20 @@ export class KernelProver {
    * @param txRequest - The authenticated transaction request object.
    * @param executionResult - The execution result object containing nested executions and preimages.
    * @param profile - Set true to profile the gate count for each circuit
-   * @param dryRun - Set true to skip the IVC proof generation (only simulation is run). Useful for profiling gate count without proof gen.
    * @returns A Promise that resolves to a KernelProverOutput object containing proof, public inputs, and output notes.
    * TODO(#7368) this should be refactored to not recreate the ACIR bytecode now that it operates on a program stack
    */
   async prove(
     txRequest: TxRequest,
     executionResult: PrivateExecutionResult,
-    { simulate, skipFeeEnforcement, profile, dryRun }: ProvingConfig = {
+    { simulate, skipFeeEnforcement, profile }: ProvingConfig = {
       simulate: false,
       skipFeeEnforcement: false,
       profile: false,
-      dryRun: false,
     },
   ): Promise<PrivateKernelSimulateOutput<PrivateKernelTailCircuitPublicInputs>> {
-    if (simulate && profile) {
-      throw new Error('Cannot simulate and profile at the same time');
-    }
-
-    simulate = simulate || this.fakeProofs;
+    const skipWitnessGeneration = simulate && !profile;
```

Comment:
I would call this `generateWitnesses`, since simulation is a completely different process